### PR TITLE
Configure ruff

### LINKED ISSUE #1022: Switch from flake8/bandit/black/isort to ruff
We've adopted ruff most everywhere else, would be good to get this repository ported over too. See https://github.com/freedomofpress/securedrop-client/pull/1885; probably worth copying that configuration to start with.

### PR DESCRIPTION
**Pull request by micahflee**
## Status

Ready for review

## Description of Changes

Fixes #1022.

This replaces flake8, bandit, black, and isort with ruff!

Many of the things that ruff flagged were `assert` statements in `validate_config.py`, and I already fixed those in #1025. So rather than fixing them again here, I merged that branch into this branch. So please review and merge #1025 first.

I started copying the ruff config from https://github.com/freedomofpress/securedrop-client/pull/1885, but then I paired it down to only include what's needed for this repo.

I added `S108` (probably insecure usage of /tmp) to the ignore list, as it was being triggered just on this code in `sdw_updater/Updater.py`:

```py
# We use a hardcoded temporary directory path in dom0. As dom0 is not
# a multi-user environment, we can safely assume that only the Updater is
# managing that filepath. Later on, we should consider porting the check-migration
# logic to leverage the Qubes Python API.
MIGRATION_DIR = "/tmp/sdw-migrations"  # nosec
```

I also added a per-file-ignore for `SIM115` (use context handler for opening files) for `sdw_util/Util.py` specifically for the `obtain_lock` and `can_obtain_lock` functions. This check wants you to open files using the `with` statement so they automatically close, but in this case these functions open files and then return the file handle. So I think making an exception is fine.

Though I did notice a potential small bug. In `can_obtain_lock`, the lock file is opened temporarily, just to see if it can be opened, but I don't think it's ever closed since it returns a bool instead of the file handle -- though maybe this is fine and I just don't understand how `fcntl.lockf` works:

```py
def can_obtain_lock(basename):
    """
    We temporarily obtain a shared, nonblocking lock to a lockfile to determine
    whether the associated process is currently running. Returns True if it is
    safe to continue execution (no lock conflict), False if not.

    `basename` is the basename of a lockfile situated in the LOCK_DIRECTORY.
    """
    lock_file = os.path.join(LOCK_DIRECTORY, basename)
    try:
        lh = open(lock_file)
    except FileNotFoundError:
        # Process may not have run during this session, safe to continue
        return True

    try:
        # Obtain a nonblocking, shared lock
        fcntl.lockf(lh, fcntl.LOCK_SH | fcntl.LOCK_NB)
    except OSError:
        sdlog.error(LOCK_ERROR.format(lock_file))
        return False

    return True
```

### CODE REVIEW

File: files/sdw-admin.py
Comment (rocodes) [inline files/sdw-admin.py:72]: wouldn't we want `check=True` here in order to trigger the exception below if the update is unsuccessful?

Comment (rocodes) [inline files/sdw-admin.py:72]: We might want `check=True` or another subprocess method (`subprocess.check_call` ?) if the intention is to error if the grub2-zen-pvh installation doesn't work. (This isn't introduced by your PR, but ruff is making it noticeable.)

(Also, not really your problem since it was preexisting, but if you're in here anyway and open to random nits, I just noticed a typo in the error message on like 74: "Error installing grub2-xen-pvah" should be "Error installing grub2-xen-pvh", no "a".)

File: sdw_updater/Updater.py
Comment (rocodes) [inline sdw_updater/Updater.py:355]: (this one's always gonna be weird for me because I feel like the `elif` is more readable and more like defensive coding even though it's not technically necessary. but okay, ruff.)

**Comment by rocodes** [comment]
Thanks for preparing this, this looks great. I'm going to look into the lock file issue now. We use the lockfile only in the Updater/+Notifier, eg to make sure that we're not bugging the user to run updates while they're already running updates.

**Review by rocodes — CHANGES_REQUESTED** [review]
Sorry to punt this back to you for just a [small fix](https://github.com/freedomofpress/securedrop-workstation/pull/1028/files#r1596837417) while our eyes are here - otherwise LGTM

File: files/sdw-admin.py
Comment (micahflee) [reply files/sdw-admin.py:72]: Good catch. This was in my "Automatic ruff fixes" commit, so I think was automatically edited when I ran `ruff check --fix`. And sure enough, if you look at the docs for [subprocess.run()](https://docs.python.org/3/library/subprocess.html#subprocess.run), the default is `check=False`, which means it's always been false. I'll change to `subprocess.check_call()` like the other functions in this file use.

File: sdw_updater/Updater.py
Comment (rocodes) [reply sdw_updater/Updater.py:355]: Thanks for pointing that out, it explains why this stuck out at me so much.

Sorry to nitpick, but for the sake of consistency, let's add back the rules that are specifically about readability, so that we have the same standards on all projects. 

So that would be, to my eye 

```
  # superflous-else- rules, find they hurt readability
    "RET505", "RET506", "RET507", "RET508",
  # Find contextlib.suppress() is harder to read
    "SIM105",
    # Find ternary statements harder to read
    "SIM108",
    # Using any()/all() can be harder to read
    "SIM110",   
```

**Review by rocodes — APPROVED** [review]

**Review by rocodes — APPROVED** [review]

File: Makefile
Comment (legoktm) [inline Makefile:190]: I think we also need to check `ruff format` - https://github.com/freedomofpress/securedrop-client/blob/2e2fdca7d280044e54578c5d4a6643a3207c77e5/Makefile#L34

File: files/destroy-vm.py
Comment (legoktm) [inline files/destroy-vm.py:43]: Can we have an error message here?

File: Makefile
Comment (legoktm) [inline Makefile:193]: Can we either call this `fix` or alias that to this? In the spirit of https://github.com/freedomofpress/securedrop-tooling/issues/11.

File: launcher/tests/test_updater.py
Comment (legoktm) [inline launcher/tests/test_updater.py:217]: the assert can happen outside the with statement

File: sdw_updater/Updater.py
Comment (legoktm) [inline sdw_updater/Updater.py:114]: same, can we have an error message here?

File: sdw_updater/Updater.py
Comment (legoktm) [reply sdw_updater/Updater.py:355]: Can we re-run the automated fixes now that those rules are suppressed? i.e. back out these changes and elsewhere

**Review by legoktm — CHANGES_REQUESTED** [review]
I rebased this now that the assert PR has landed and left some comments. One other thing is that the `# nosec` comments needed for bandit can now be removed.

**Comment by micahflee** [comment]
The tests are failing because of this:

>   Unable to find installation candidates for pyqt5-qt5 (5.15.13)

I'm not sure what's wrong there, so any help is appreciated.

File: poetry.lock
Comment (legoktm) [inline poetry.lock:192]: Something went wrong here, not sure why this package was upgraded. Let me back it out and see if that fixes the CI issue.

Comment (legoktm) [reply poetry.lock]: Ah yeah, that'll do it; it tries to update everything unless you use `--no-update`

**Review by legoktm — CHANGES_REQUESTED** [review]
Overall looks ready to land now, but could you rebase/squash this? I think some things landed in tests/test_vms_platform.py unintentionally.

**Review by legoktm — APPROVED** [review]
Rebased, just waiting on CI before merging

**Pull request merged**
